### PR TITLE
[SPARK-16088][SPARKR] update setJobGroup, cancelJobGroup, clearJobGroup

### PR DESCRIPTION
**Pull request by felixcheung**
## What changes were proposed in this pull request?

Updated setJobGroup, cancelJobGroup, clearJobGroup to not require sc/SparkContext as parameter.
Also updated roxygen2 doc and R programming guide on deprecations.
## How was this patch tested?

unit tests

### CODE REVIEW

**Comment by felixcheung** [comment]
@shivaram

File: R/pkg/R/sparkR.R [hunk 1]
```diff
@@ -392,47 +392,81 @@ sparkR.session <- function(
...
   callJMethod(sc, "setJobGroup", groupId, description, interruptOnCancel)
 }
 
+setJobGroup <- function(sc, groupId, description, interruptOnCancel) {
+  if (class(sc) == "jobj" && any(grepl("JavaSparkContext", getClassName.jobj(sc)))) {
+    .Deprecated("setJobGroup(groupId, description, interruptOnCancel)",
+                old = "setJobGroup(sc, groupId, description, interruptOnCancel)")
+    setJobGroup.default(groupId, description, interruptOnCancel)
+  } else {
+    # Parameter order is shifted
```
Comment (shivaram) [inline R/pkg/R/sparkR.R]: Would it be cleaner to name the variables differently ? Something like `groupIdToUse <- sc` etc.

Also just curious -- what are the semantics of calling a 4 argument function with just 3 arguments in S3 ?

File: R/pkg/R/sparkR.R [hunk 2]
```diff
@@ -392,47 +392,81 @@ sparkR.session <- function(
...
 #' @examples
 #'\dontrun{
-#' sc <- sparkR.init()
-#' cancelJobGroup(sc, "myJobGroup")
+#' sparkR.session()
+#' cancelJobGroup("myJobGroup")
 #'}
 #' @note cancelJobGroup since 1.5.0
-cancelJobGroup <- function(sc, groupId) {
+#' @method cancelJobGroup default
```
Comment (shivaram) [inline R/pkg/R/sparkR.R]: can we set the rdname here to be `cancelJobGroup` ? (and same for other ones). The reason is that with this PR doing `?cancelJobGroup` gives me a `No documentation for ‘cancelJobGroup’ in specified packages and libraries:`

File: R/pkg/R/sparkR.R (hunk 1 above)
Comment (felixcheung) [inline R/pkg/R/sparkR.R]: the 4th argument would just be "missing"
it doesn't work to call 3 arguments function with 4 arguments though.

File: R/pkg/R/sparkR.R (hunk 2 above)
Comment (felixcheung) [inline R/pkg/R/sparkR.R]: hmm, curious - roxygen2 should have made a guess of the rdname - in fact I see the rd/html generated for cancelJobGroup.

**Comment by shivaram** [comment]
LGTM. Merging this to master, branch-2.0

**Pull request closed without merging**